File: src/webapp/client/src/ts/types/vendors.ts
export interface RawVendor {
    Id              : number
    EngagementId    : number
    Name            : string
    Description     : string
    Url             : string
}

export interface RawVendorProduct {
    Id              : number
    VendorId        : number
    Name            : string
    Description     : string
    Url             : string
}

export function createEmptyVendor() : RawVendor {
    return {
        Id: -1,
        EngagementId: -1,
        Name: '',
        Description: '',
        Url: '',
    }
}

export function createEmptyVendorProduct() : RawVendorProduct {
    return {
        Id: -1,
        VendorId: -1,
        Name: '',
        Description: '',
        Url: '',
    }
}

File: src/shared/golang/backend/vendors/vendors.go
package vendors

import (
	"github.com/jmoiron/sqlx"
)

type Vendor struct {
	Id           int64  `db:"id"`
	EngagementId int64  `db:"engagement_id"`
	Name         string `db:"name"`
	Description  string `db:"description"`
	Url          string `db:"url"`
}

type VendorProduct struct {
	Id          int64  `db:"id"`
	VendorId    int64  `db:"vendor_id"`
	Name        string `db:"name"`
	Description string `db:"description"`
	Url         string `db:"url"`
}

type VendorManager struct {
	db *sqlx.DB
}

func CreateVendorManager(db *sqlx.DB) *VendorManager {
	return &VendorManager{
		db: db,
	}
}
